gridstatus/spp.py

```diff
@@ -672,20 +672,26 @@ def _finalize_spp_df(self, df, market, location_type, verbose=False):
         df = df.reset_index(drop=True)
         return df
 
-    @support_date_range("DAY_START")
-    def get_lmp_real_time_weis(self, date, verbose=False):
+    @support_date_range("5_MIN")
+    def get_lmp_real_time_weis(self, date, end=None, verbose=False):
         """Get LMP data for real time WEIS
 
         Args:
-            date: date to get data for
+            date: date to get data for. if end is not provided, will get data for
+                5 minute interval that date is in.
+            end: end date
+            verbose: print url
         """
+        # if no end, find nearest 5 minute interval end
+        # to use
+        if end is None:
+            # round date up to nearest 5 minutes
+            end = date + pd.Timedelta(minutes=5) - pd.Timedelta(
```

Comment:
We might want to handle the case where `date="latest"`. Currently we get an error here.

Reply:
We could add the following before the `if end is None` conditional:

```py
if date == "latest":
    date = pd.Timestamp.now(tz=self.default_timezone).ceil("5min")
```

Reply:
there is a specific url for latest, so i implemented that